### PR TITLE
[Enhancement] Add statistic collect parallel num in session

### PR DESCRIPTION
**Pull request by Seaven**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 14 / 17 (82.35%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|8|10|80.00%|
|:large_blue_circle:|com/starrocks/statistic/StatisticUtils.java|4|5|80.00%|
|:large_blue_circle:|com/starrocks/statistic/StatisticExecutor.java|2|2|100.00%|

**Review by kangkaisen — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java [hunk 1]
```diff
@@ -204,6 +204,8 @@ public class SessionVariable implements Serializable, Writable, Cloneable {
     public static final String ENABLE_OPTIMIZER_TRACE_LOG = "enable_optimizer_trace_log";
     public static final String JOIN_IMPLEMENTATION_MODE = "join_implementation_mode";
 
+    public static final String STATISTIC_COLLECT_PARALLEL = "statistic_collect_parallel";
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:207]: parallelism?

File: fe/fe-core/src/main/java/com/starrocks/statistic/StatisticUtils.java [hunk 2]
```diff
@@ -31,8 +31,12 @@ public static ConnectContext buildConnectContext() {
         // but QeProcessorImpl::reportExecStatus will check query id,
         // So we must disable report query status from BE to FE
         context.getSessionVariable().setReportSuccess(false);
-        // Always use 1 parallel to avoid affect normal query
-        context.getSessionVariable().setParallelExecInstanceNum(1);
+        int parallel = context.getSessionVariable().getStatisticCollectParallel();
+        if (null != ConnectContext.get()) {
+            parallel = ConnectContext.get().getSessionVariable().getStatisticCollectParallel();
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/statistic/StatisticUtils.java:36]: Why only parallel needs to judge ConnectContext

Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/statistic/StatisticUtils.java:36]: Other variable don't need modify

**Comment by Seaven** [comment]
run starrocks_fe_unittest

**Comment by Seaven** [comment]
run starrocks_fe_unittest

**Comment by Seaven** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 13 / 17 (76.47%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/statistic/StatisticExecutor.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|8|10|80.00%|
|:large_blue_circle:|com/starrocks/statistic/StatisticUtils.java|4|5|80.00%|

File: fe/fe-core/src/main/java/com/starrocks/statistic/StatisticUtils.java [hunk 3]
```diff
@@ -31,8 +31,12 @@ public static ConnectContext buildConnectContext() {
         // but QeProcessorImpl::reportExecStatus will check query id,
         // So we must disable report query status from BE to FE
         context.getSessionVariable().setReportSuccess(false);
-        // Always use 1 parallel to avoid affect normal query
-        context.getSessionVariable().setParallelExecInstanceNum(1);
+        int parallel = context.getSessionVariable().getStatisticCollectParallelism();
+        if (null != ConnectContext.get()) {
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/statistic/StatisticUtils.java:35]: When ConnectContext.get() will appear Null， and what different with context default session variable and ConnectContext.get().getSessionVariable()?

Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/statistic/StatisticUtils.java:35]: AutoCollector may null, if you only use default session variable, you need set the variable by `set global`

File: fe/fe-core/src/main/java/com/starrocks/statistic/StatisticUtils.java [hunk 4]
```diff
@@ -31,8 +31,12 @@ public static ConnectContext buildConnectContext() {
...
+            parallel = ConnectContext.get().getSessionVariable().getStatisticCollectParallelism();
+        }
+        context.getSessionVariable().setParallelExecInstanceNum(parallel);
+        context.getSessionVariable().setPipelineDop(1);
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/statistic/StatisticUtils.java:39]: Why set to 1?

Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/statistic/StatisticUtils.java:40]: Avoid pipelien auto ajust instanceNum later

**Review by Youngwb — DISMISSED** [review]

**Comment by Seaven** [comment]
run starrocks_fe_unittest

**Review by HangyuanLiu — DISMISSED** [review]

**Review by kangkaisen — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Pull request merged**